maint/mocked/based-notifyfenced.c

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright 2019 the Pacemaker project contributors
+ *
+ * The version control history for this file may have further details.
+ *
+ * Licensed under the GNU General Public License version 2 or later (GPLv2+).
```

Comment:
Is it really that painful to copy and paste the current wording as documented in Pacemaker Development? This is close enough that I'll merge it, but I don't understand what's so objectionable about the existing recommendation.

The wording has some rational basis. The GPL requires that anyone who modifies the original source "conspicuously and appropriately publish on each copy an appropriate copyright notice and disclaimer of warranty". That doesn't apply to the original code, and I doubt most people today care about the warranty disclaimer, but it's in the spirit of the GPL to include it, for ease of compliant modification.

Reply:
Warranty disclaimer is expressed in the license file already.

Moreover, Is there any country one is held responsible for whatever broken
code?  That's in steep contrast with "nontrivial code always contains a single
bug or more", meaning the prisons wouldn't suffice :-)

I can understand comparability of single line (words) vs. single line (SPDX),
and that FSFAP is not established in the project (and doing so can wait),
but cargo cult + said cognitive load, yes, that's some collateral pain we shall
alleviate, when sensible (again, no stirring for stirring).

Reply:
> Warranty disclaimer is expressed in the license file already.

That's not the relevant portion. The GPL requires all modifications of GPL'd code to "publish on each copy" (*separately* from including a copy of the license itself) a warranty disclaimer, along with the copyright notice. Since there is no physical box or CD here, the "copy" that includes the copyright notice and warranty disclaimer is the file itself. Technically, to comply with the license, anyone who modifies this file after you would have to add the disclaimer. That's why it makes sense to put it there from the beginning.

> Moreover, Is there any country one is held responsible for whatever broken
> code? That's in steep contrast with "nontrivial code always contains a single
> bug or more", meaning the prisons wouldn't suffice :-)

Yes, the warranty disclaimer originates with long-ago lawsuits. Whether it is an appropriate remedy for the purpose or not is another question, but again, the GPL requires it.
 
> I can understand comparability of single line (words) vs. single line (SPDX),

While I appreciate your desire for brevity, the more fundamental point here is that setting and enforcing style policies is a proper function of any software project. When I submit PRs against sosreport, I follow their style guidelines, not pacemaker's. It's reasonable to disagree about the merits of a style guideline, but it makes no sense to just charge ahead with your own preferences contrary to the guidelines. Guidelines can be changed, but such changes should be proposed for discussion in a dedicated PR.

> and that FSFAP is not established in the project (and doing so can wait),
> but cargo cult + said cognitive load, yes, that's some collateral pain we shall
> alleviate, when sensible (again, no stirring for stirring).

I've noticed a tendency for you to jump to pejorative terms such as "cargo culting", "bikeshedding", and "micromanaging" for routine, reasonable requests. "Cargo culting" refers to parroting an existing practice without consideration of its rationale. Not only do I have reasons for my recommendations, but I've tried to explain them to you in detail. You may not find them convincing, but that does not contravene their existence.

In any case, all that is required in this PR is to rebase on current master and resolve the merge conflict.